### PR TITLE
New Command: `slcli user grant-access`

### PR DESCRIPTION
**Pull request by BrianSantivanez**
Issue: https://github.com/softlayer/softlayer-python/issues/1816
Observations: `slcli user grant-access` command and unit tests were added
```
slcli user grant-access 8344222 --hardware 1532729 --virtual 132364928
Grant to access to hardware: 1532729
Grant to access to virtual guest: 132364928
```

### CODE REVIEW

File: SoftLayer/CLI/user/grant_access.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+"""User grant access to devices."""
```
Comment (allmightyspiff) [inline SoftLayer/CLI/user/grant_access.py:1]:
```suggestion
"""Grants a user access to a given device"""
```

File: SoftLayer/CLI/user/grant_access.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+@click.command(cls=SLCommand, )
+@click.argument('identifier')
+@click.option('--hardware', help="Hardware ID")
+@click.option('--virtual', help="Virtual Guest ID")
+@click.option('--dedicated', help="dedicated host ID")
+@environment.pass_env
+def cli(env, identifier, hardware, virtual, dedicated):
+    """Grant access from a user to an specific device.
```
Comment (allmightyspiff) [inline SoftLayer/CLI/user/grant_access.py:18]:
```suggestion
    """Grants a user access to a given device.
```

File: SoftLayer/CLI/user/grant_access.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    Example: slcli user grant-access 123456 --hardware 123456789
+    """
+
+    mgr = SoftLayer.UserManager(env.client)
+    result = False
+    if hardware:
+        result = mgr.grant_hardware_access(identifier, hardware)
+        if result:
+            click.secho("Grant to access to hardware: %s" % hardware, fg='green')
```
Comment (allmightyspiff) [inline SoftLayer/CLI/user/grant_access.py:28]:
```suggestion
            click.secho(f"User {identifier}  has been given access to hardware {hardware} ", fg='green')
```

I'd like to start using [f strings](https://fstring.help) anything we need variable substitution now that we only support py3.7+.  And the message needed to be cleaned up a little.

Use the same format on the other 2 messages below as well.

File: SoftLayer/CLI/user/grant_access.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    if dedicated:
+        result = mgr.grant_dedicated_access(identifier, dedicated)
+        if result:
+            click.secho("Grant to access to dedicated host: %s" % dedicated, fg='green')
+
+    if not result:
+        raise SoftLayer.exceptions.SoftLayerError('You need argument a hardware, virtual or dedicated identifier.\n'
+                                                  'E.g slcli user grant-access 123456 --hardware 91803794\n'
+                                                  '    slcli user grant-access 123456 --dedicated 91803793\n'
+                                                  '    slcli user grant-access 123456 --virtual 91803792')
```
Comment (allmightyspiff) [inline SoftLayer/CLI/user/grant_access.py:44]:
```suggestion
    if not result:
        raise SoftLayer.exceptions.SoftLayerError('A device option is required.\n'
                                                  'E.g slcli user grant-access 123456 --hardware 91803794\n'
                                                  '    slcli user grant-access 123456 --dedicated 91803793\n'
                                                  '    slcli user grant-access 123456 --virtual 91803792')
```

File: tests/CLI/modules/user_tests.py [hunk 5]
```diff
@@ -345,6 +345,18 @@ def test_devices_access(self):
...
+        self.assert_no_fail(result)
+
+    def test_grant_access_virtual(self):
+        result = self.run_command(['user', 'grant-access', '123456', '--virtual', '987456'])
+        self.assert_no_fail(result)
+
+    def test_grant_access_dedicated(self):
+        result = self.run_command(['user', 'grant-access', '123456', '--dedicated', '369852'])
+        self.assert_no_fail(result)
+
```
Comment (allmightyspiff) [inline tests/CLI/modules/user_tests.py:359]: Add a unit test for when a user doesn't have any device options to test the exception.

**Review by allmightyspiff — CHANGES_REQUESTED** [review]
Looks good, just a few minor changes needed thanks.

File: tests/CLI/modules/user_tests.py (hunk 5 above)
Comment (BrianSantivanez) [reply tests/CLI/modules/user_tests.py:359]: The negative unit test was added

File: SoftLayer/CLI/user/grant_access.py (hunk 4 above)
Comment (BrianSantivanez) [reply SoftLayer/CLI/user/grant_access.py:44]: Done

File: SoftLayer/CLI/user/grant_access.py (hunk 3 above)
Comment (BrianSantivanez) [reply SoftLayer/CLI/user/grant_access.py:28]: The messages were updated using `f strings`

File: SoftLayer/CLI/user/grant_access.py (hunk 2 above)
Comment (BrianSantivanez) [reply SoftLayer/CLI/user/grant_access.py:18]: Done

File: SoftLayer/CLI/user/grant_access.py (hunk 1 above)
Comment (BrianSantivanez) [reply SoftLayer/CLI/user/grant_access.py:1]: Done

**Review by caberos — APPROVED** [review]
LGTM

**Comment by edsonarios** [comment]
Screenshots of the test result
image

**Review by edsonarios — APPROVED** [review]
Looks good to me

**Review by allmightyspiff — APPROVED** [review]

**Pull request merged**